File: Android/app/src/main/java/com/simplex/coin2credit/TransactionDao.java
package com.simplex.coin2credit;

import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Delete;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.Query;

import java.util.List;

@Dao
public interface TransactionDao {
    @Query("SELECT * FROM transactionDB")
    List<Transaction> getAll();

    @Query("SELECT * FROM transactionDB where date LIKE :date")
    Transaction findByDate(String date);

    @Query("SELECT COUNT(*) from transactionDB")
    int countTransactions();

    @Insert
    void insertAll(Transaction... transactions);

    @Delete
    void delete(Transaction transaction);
}

File: Android/app/src/main/java/com/simplex/coin2credit/database/TransactionDao.kt
package com.simplex.coin2credit.database

import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.Query

@Dao
interface TransactionDao {
    @get:Query("SELECT * FROM transactionDB")
    val all: List<Transaction>

    @Query("SELECT * FROM transactionDB where date LIKE :date")
    fun findByDate(date: String): Transaction

    @Query("SELECT COUNT(*) from transactionDB")
    fun countTransactions(): Int

    @Insert
    fun insertAll(vararg transactions: Transaction)

    @Delete
    fun delete(transaction: Transaction)
}
